### PR TITLE
ATOR-109 Rename Tor to Anon

### PR DESCRIPTION
**Pull request by floodcode**
Renamed entities:
- config `torrc` -> `anonrc`
- binary `tor` -> `anon`
- binary `tor-gencert` -> `anon-gencert`
- binary `tor-print-ed-signing-cert` -> `anon-print-ed-signing-cert`
- binary `tor-resolve` -> `anon-resolve`
- binary `torify` -> `anonify`
- library `libtor` -> `libanon`
- debian package `tor` -> `anon`
- state directory `tor` -> `anon`
- config directory `tor` -> `anon`
- docker image `ator-protocol` -> `anon-dev`

Other changes:
- updated version to `anon-0.4.9.1-dev`

Tested:
- local build + run
- docker image build + run
- running nyx over control port and control socket

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -303,7 +303,7 @@ Changes in version 0.4.8.1-alpha - 2023-06-01
     - Implement proposal 327 (Proof-Of-Work). This is aimed at thwarting
       introduction flooding DoS attacks by introducing a dynamic Proof-Of-Work
       protocol that occurs over introduction circuits. This introduces several
-      torrc options prefixed with "HiddenServicePoW" in order to control this
+      anonrc options prefixed with "HiddenServicePoW" in order to control this
```
Comment (yumirkov) [inline ChangeLog:306]: Do not think it should be renamed

File: ReleaseNotes [hunk 2]
```diff
@@ -165,7 +165,7 @@ Changes in version 0.4.8.4 - 2023-08-23
     - Implement proposal 327 (Proof-Of-Work). This is aimed at thwarting
       introduction flooding DoS attacks by introducing a dynamic Proof-Of-Work
       protocol that occurs over introduction circuits. This introduces several
-      torrc options prefixed with "HiddenServicePoW" in order to control this
```
Comment (yumirkov) [inline ReleaseNotes:168]: The same as ChangeLog

File: configure.ac [hunk 3]
```diff
@@ -18,7 +18,7 @@ AC_DEFINE_UNQUOTED([CONFIG_FLAGS], ["$configure_flags"], [Flags passed to config
 # version number changes.  Tor uses it to make sure that it
 # only shuts down for missing "required protocols" when those protocols
 # are listed as required by a consensus after this date.
-AC_DEFINE(APPROX_RELEASE_DATE, ["2023-08-23"], # for 0.4.9.0-alpha-dev
+AC_DEFINE(APPROX_RELEASE_DATE, ["2023-12-30"], # for 0.4.10.0-alpha-dev
```
Comment (yumirkov) [inline configure.ac:21]: Rollback this please

File: contrib/win32build/anon-mingw.nsi.in [hunk 4]
```diff
@@ -8,11 +8,11 @@
 !include "LogicLib.nsh"
 !include "FileFunc.nsh"
 !insertmacro GetParameters
-!define VERSION "0.4.9.0-alpha-dev"
+!define VERSION "0.4.10.0-alpha-dev"
```
Comment (yumirkov) [inline contrib/win32build/anon-mingw.nsi.in:11]: I believe it should also be reverted

Comment (floodcode) [reply contrib/win32build/anon-mingw.nsi.in]: It is auto generated with `./scripts/maint/update_versions.py`, same as `APPROX_RELEASE_DATE` and comments, will update `AC_INIT()` is `configure.ac` and regenerate

File: ChangeLog (hunk 1 above)
Comment (floodcode) [reply ChangeLog]: Reverted

File: ReleaseNotes (hunk 2 above)
Comment (floodcode) [reply ReleaseNotes]: Reverted

File: contrib/win32build/anon-mingw.nsi.in (hunk 4 above)
Comment (floodcode) [reply contrib/win32build/anon-mingw.nsi.in]: Updated to `0.4.9.0-alpha-dev`

File: configure.ac (hunk 3 above)
Comment (floodcode) [reply configure.ac]: Reverted

File: docker/Dockerfile [hunk 5]
```diff
@@ -1,42 +1,42 @@
...
 # as they are evaluated at build time and cannot be changed using only ENV variables.
-ENV TOR_ORPORT=${BUILD_TOR_ORPORT}
-ENV TOR_DIRPORT=${BUILD_TOR_DIRPORT}
+ENV ANON_ORPORT=${BUILD_ANON_ORPORT}
+ENV ANON_DIRPORT=${BUILD_ANON_DIRPORT}
 
 # Sets a default nickname.
 # To set email field pass CONTACT_EMAIL env variable.
-ENV TOR_NICKNAME=ATORv4
+ENV ANON_NICKNAME=ATORv4
```
Comment (yumirkov) [inline docker/Dockerfile:27]: How about change nickname as well

Comment (floodcode) [reply docker/Dockerfile:27]: Makes sense

File: docker/examples/da/Dockerfile [hunk 6]
```diff
@@ -2,26 +2,26 @@ FROM debian:bookworm
 
 ARG DEBIAN_FRONTEND=noninteractive
 
-ARG ATOR_VER="8a729c50a87974b78822227c14c03f3fb6be514e"
-ARG ATOR_REPO="https://github.com/ATOR-Development/ator-protocol.git"
+ARG ANON_VER="8a729c50a87974b78822227c14c03f3fb6be514e"
```
Comment (yumirkov) [inline docker/examples/da/Dockerfile:5]: Will this tag work?

File: configure.ac [hunk 7]
```diff
@@ -18,7 +18,7 @@ AC_DEFINE_UNQUOTED([CONFIG_FLAGS], ["$configure_flags"], [Flags passed to config
 # version number changes.  Tor uses it to make sure that it
 # only shuts down for missing "required protocols" when those protocols
 # are listed as required by a consensus after this date.
-AC_DEFINE(APPROX_RELEASE_DATE, ["2023-08-23"], # for 0.4.9.0-alpha-dev
+AC_DEFINE(APPROX_RELEASE_DATE, ["2024-01-04"], # for 0.4.9.0-alpha-dev
```
Comment (yumirkov) [inline configure.ac:21]: What about date rollback?

**Review by yumirkov — APPROVED** [review]

**Pull request merged**